pycheribuild/projects/cross/freertos.py

```diff
@@ -76,22 +94,23 @@ def __init__(self, config: CheriConfig):
         # Override that with no and set the appopriate flags here.
         self.make_args.set(NIX_ENV="no")
 
-        # Only build 64-bit FreeRTOS as cheribuild currently only supports building
-        # for RV64
-        self.make_args.set(RISCV_XLEN="64")
-
         # Set sysroot Makefile arg to pick up libc
         self.make_args.set(SYSROOT=str(self.sdk_sysroot))
 
         # Add compiler-rt location to the search path
         # self.make_args.set(LDFLAGS="-L"+str(self.compiler_resource / "lib"))
 
-        if self.target_info.target.is_cheri_purecap():
-            # CHERI-RISC-V sophisticated Demo with more advanced device drivers
-            # and currently only runs on FPGA-GFE, purecap
-            self.supported_freertos_demos.append("RISC-V_Galois_P1")
-            self.supported_demo_apps["RISC-V_Galois_P1"] = ["main_blinky", "main_netboot"]
+        if self.demo == "RISC-V_Galois_P1":
+            if "rv32" in self.demo_bsp:
+                self.make_args.set(XARCH="32")
+            else:
+                self.make_args.set(XARCH="64")
 
+        else:  # RISC-V-Generic
+            # Only build 64-bit FreeRTOS for RISC-V-Generic
```

Comment:
Do one of:

* assert
* leave it and let the target fail (as you do for non-CHERI RISC-V_Galois_P1 demos)
* cheribuild magic so the incompatible combinations don't exist as targets